cmake/SetupCxxFlags.cmake

```diff
@@ -34,14 +34,14 @@ if(NOT "${OVERRIDE_ARCH}" STREQUAL "OFF")
       # The -mno-avx512f flag is necessary to avoid a Blaze 3.8 bug. The flag
       # should be re-enabled when we can insist on Blaze 3.9 which will include
       # a fix that allows this vectorization flag again.
-      $<$<COMPILE_LANGUAGE:CXX>:-march=${OVERRIDE_ARCH} -mno-avx512f>)
+      $<$<COMPILE_LANGUAGE:C,CXX,Fortran>:-march=${OVERRIDE_ARCH} -mno-avx512f>)
```

Comment:
It looks like the oldest CMake version we support doesn't like this syntax (that's the CI error). It wants:
```
$<$<COMPILE_LANGUAGE:C>:-march=${OVERRIDE_ARCH} -mno-avx512f>
$<$<COMPILE_LANGUAGE:CXX>:-march=${OVERRIDE_ARCH} -mno-avx512f>
$<$<COMPILE_LANGUAGE:Fortran>:-march=${OVERRIDE_ARCH} -mno-avx512f>)
```
I think. But please verify that still fixes the issues you're seeing on your ShapeMap changes. You'll need to make the same change below on line 44 and in the SetupPic file as well (both for the PR and for testing your branch!)